triton-vm/src/stark.rs

```diff
@@ -2575,173 +2577,49 @@ pub(crate) mod tests {
-            quotient_segment_codewords_new
-        );
-        prop_assert_eq!(
-            quotient_segment_polynomials_old,
-            quotient_segment_polynomials_new
-        );
-    }
-
-    fn compute_quotient_segments_old(
-        main_polynomials: ArrayView1<Polynomial<BFieldElement>>,
-        aux_polynomials: ArrayView1<Polynomial<XFieldElement>>,
-        trace_domain: ArithmeticDomain,
-        quotient_domain: ArithmeticDomain,
-        fri_domain: ArithmeticDomain,
-        challenges: &Challenges,
-        quotient_weights: &[XFieldElement],
-    ) -> (Array2<XFieldElement>, Array1<Polynomial<XFieldElement>>) {
-        let mut main_quotient_domain_codewords =
-            Array2::<BFieldElement>::zeros([quotient_domain.length, MasterMainTable::NUM_COLUMNS]);
-        Zip::from(main_quotient_domain_codewords.axis_iter_mut(Axis(1)))
-            .and(main_polynomials.axis_iter(Axis(0)))
-            .for_each(|codeword, polynomial| {
-                Array1::from_vec(quotient_domain.evaluate(&polynomial[()])).move_into(codeword);
-            });
-        let mut aux_quotient_domain_codewords =
-            Array2::<XFieldElement>::zeros([quotient_domain.length, MasterAuxTable::NUM_COLUMNS]);
-        Zip::from(aux_quotient_domain_codewords.axis_iter_mut(Axis(1)))
-            .and(aux_polynomials.axis_iter(Axis(0)))
-            .for_each(|codeword, polynomial| {
-                Array1::from_vec(quotient_domain.evaluate(&polynomial[()])).move_into(codeword);
-            });
-
-        let quotient_codeword = all_quotients_combined(
-            main_quotient_domain_codewords.view(),
-            aux_quotient_domain_codewords.view(),
-            trace_domain,
-            quotient_domain,
-            challenges,
-            quotient_weights,
-        );
-        let quotient_codeword = Array1::from(quotient_codeword);
-        let quotient_segment_polynomials =
-            Stark::interpolate_quotient_segments(quotient_codeword, fri_domain);
-        let quotient_segment_codewords =
-            Stark::fri_domain_segment_polynomials(quotient_segment_polynomials.view(), fri_domain);
-
-        (quotient_segment_codewords, quotient_segment_polynomials)
-    }
-
     #[proptest]
     fn polynomial_segments_cohere_with_originating_polynomial(
         #[strategy(2_usize..8)] log_trace_length: usize,
-        #[strategy(1_usize..#log_trace_length.min(3))] log_num_segments: usize,
+        #[strategy(2_usize..=#log_trace_length.min(4))] log_num_cosets: usize,
         #[strategy(1_usize..6)] log_expansion_factor: usize,
-        #[strategy(vec(arb(), (1 << #log_num_segments) * (1 << #log_trace_length)))]
+        #[strategy(vec(arb(), (1 << #log_num_cosets) * (1 << #log_trace_length)))]
         coefficients: Vec<XFieldElement>,
         #[strategy(arb())] random_point: XFieldElement,
     ) {
-        let polynomial = Polynomial::new(coefficients);
+        const NUM_SEGMENTS: usize = 4;
```

Comment:
This test does not verify the genericity of the function `segmentify`, which was one of the purposes for the re-write. I would suggest writing a generic function body that `prop_assert`s the right properties, and then calling that function several times with various static parameters from the proptest.